File: cap10/src/algoritmos/CifradoCesar.py
BASE_MIN = ord('a')
BASE_MAY = ord('A')
LONG_ABC = 26

def cifrar(texto, despl):
    cifrado = ""
    for letra in texto:
        cifrado += cifrar_letra(letra, despl)
    return cifrado
    
def cifrar_letra(letra, despl):
    if letra.isupper():
        base = BASE_MAY
    elif letra.islower():
        base = BASE_MIN
    else:
        return letra # no ciframos las no letras
    return chr((ord(letra) - base + despl) % LONG_ABC + base)

for i in range (0, 27):
    print(str(i) + "> " + cifrar("Buenos dias", i))

print(cifrar("Buenos dias", 13))
print(cifrar(cifrar("Buenos dias", 13), 13))    

File: cap10/src/algoritmos/CifradoCesar.java
package algoritmos;

public class CifradoCesar {

	private static final int BASE_MAY = (int)'A';
	private static final int BASE_MIN = (int)'a';
	private static final int LONG_ABC = 26;

	public static String cifrar(String texto, int despl) {
		String cifrado = "";
		for (char letra : texto.toCharArray()) {
			cifrado += cifrar(letra, despl);
		}
		return cifrado;
	}
	
	private static char cifrar(char letra, int despl) {
		int base;
		if (Character.isUpperCase(letra)) {
			base = BASE_MAY;
		} else if (Character.isLowerCase(letra)) {
			base = BASE_MIN;
		} else {
			return letra;
		}
		return (char)(((int)letra - base + despl) % LONG_ABC + base);
	}
	
	public static void main(String[] args) {
		for(int i = 0; i < 27; i++) {
			System.out.println(i + "> " + cifrar("Buenos dias", i));
		}

		System.out.println(cifrar("Buenos dias", 13));
		System.out.println(cifrar(cifrar("Buenos dias", 13), 13));
	}
}